homeassistant/components/climate/reproduce_state.py

```diff
@@ -0,0 +1,105 @@
+"""Module that groups code required to handle state restore for component."""
+import asyncio
+from typing import Iterable, Optional
+
+from homeassistant.const import (
+    ATTR_TEMPERATURE, SERVICE_TURN_OFF,
+    SERVICE_TURN_ON, STATE_OFF, STATE_ON)
+from homeassistant.core import Context, State
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.loader import bind_hass
+
+
+async def _async_reproduce_states(hass: HomeAssistantType,
+                                  state: State,
+                                  context: Optional[Context] = None) -> None:
+    """Reproduce component states."""
+    from . import (
+        ATTR_AUX_HEAT,
+        ATTR_AWAY_MODE,
+        ATTR_TARGET_TEMP_HIGH,
+        ATTR_TARGET_TEMP_LOW,
+        ATTR_HOLD_MODE,
+        ATTR_OPERATION_MODE,
+        ATTR_SWING_MODE,
+        ATTR_HUMIDITY,
+        SERVICE_SET_AWAY_MODE,
+        SERVICE_SET_AUX_HEAT,
+        SERVICE_SET_TEMPERATURE,
+        SERVICE_SET_HOLD_MODE,
+        SERVICE_SET_OPERATION_MODE,
+        SERVICE_SET_SWING_MODE,
+        SERVICE_SET_HUMIDITY,
+        DOMAIN,
+    )
+
+    from .ecobee import (
```

Comment:
Common component modules shouldn't be depend on platforms.

Reply:
It's an architecture decision. We don't want to have that dependency. At least so far we haven't.